numpy/core/tests/test_ufunc.py

```diff
@@ -2002,6 +2002,23 @@ def test_inplace_fancy_indexing(self):
         # Test multiple output ufuncs raise error, gh-5665
         assert_raises(ValueError, np.modf.at, np.arange(10), [1])
 
+        # Test ufuncs with non-trivial signature raise a TypeError
+        a = np.ones((2, 2, 2))
+        b = np.ones((1, 2, 2))
+        assert_raises(TypeError, np.matmul.at, a, [0], b)
+
+        a = np.array([[[1, 2], [3, 4]]])
+        assert_raises(TypeError, np.linalg._umath_linalg.det.at, a, [0])
+
+        # Ufunc with None signature should work as intended
```

Comment:
I guess we should have tests for this already?  So I think you can remove them again.
If you think they add something, just move them above to that long list for now...